crates/hir-ty/src/infer/pat.rs

```diff
@@ -262,13 +263,18 @@ impl<'a> InferenceContext<'a> {
                 if let &Some(slice_pat_id) = slice {
                     let rest_pat_ty = match expected.kind(Interner) {
                         TyKind::Array(_, length) => {
-                            let length = match length.data(Interner).value {
+                            let len = match length.data(Interner).value {
                                 ConstValue::Concrete(ConcreteConst {
-                                    interned: ConstScalar::Usize(length),
-                                }) => length.checked_sub((prefix.len() + suffix.len()) as u64),
+                                    interned: ConstScalar::UInt(len),
+                                }) => len.checked_sub((prefix.len() + suffix.len()) as u128),
                                 _ => None,
                             };
-                            TyKind::Array(elem_ty.clone(), crate::consteval::usize_const(length))
+                            TyKind::Array(
+                                elem_ty.clone(),
+                                intern_const_scalar(
```

Comment:
shouldn't this explicitly specify type usize instead of u128?

Reply:
@flodiebold this requires adding `ConstScalar::Usize` variant along with `ConstScalar::UInt`, and it feels kinda redundant.

Reply:
Maybe I'm missing something, but I think it should just be `intern_const_scalar_with_type(..., TyBuilder::usize())`? This `intern_const_scalar` is not a good function to have, I think.